ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/retriever/RecoveringSidecarRetriever.java

```diff
@@ -212,14 +230,22 @@ public synchronized void addSidecar(DataColumnSidecar sidecar) {
         existingSidecarsByColIdx.put(sidecar.getIndex(), sidecar);
         if (existingSidecarsByColIdx.size() >= recoverColumnCount) {
           // TODO: Make it asynchronously as it's heavy CPU operation
-          recover();
-          recoveryComplete();
+          try (MetricsHistogram.Timer ignored =
+              dataAvailabilityReconstructionTimeSeconds.startTimer()) {
+            recover();
+            recoveryComplete();
+          } catch (final Throwable t) {
+            LOG.error(
+                "Failed to reconstruct data column sidecar {}", sidecar::toLogString, () -> t);
+          }
         }
       }
     }
 
     private void recoveryComplete() {
       recovered = true;
+      totalDataAvailabilityReconstructedColumns.inc(
```

Comment:
Move `promisesByColIdx.values().stream().mapToInt(List::size).sum()` to variable and reuse, please